### PR TITLE
Disabled the trade panel if Optimism is not connected

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
Disable the trade panel if the wallet is not connected or the network is unsupported

## Related issue
#1405

## Motivation and Context
Make sure the user can only interact with the trade panel when connecting the optimism mainnet or optimism goerli.

## How Has This Been Tested?
Case 1: No wallet connected
1. Visit the futures page
2. The trade panel is replaced by the message `connect wallet`
3. Click on the `connect wallet`
4. The trade panel is visible after connecting the wallet

Case 2: Connected to Polygon
1. Visit the futures page 
2. The trade panel is replaced by the message `Switch to L2`
3. Click on the `Switch to L2`
4. The trade panel is visible after switching to L2

## Screenshots (if appropriate):
image
image

### CODE REVIEW

**Comment by platschi** [comment]
Works fine so far, just one small nitpick (button colors should be responsive to theme:)

Screen Shot 2022-09-23 at 09 02 28

**Review by avclarke — COMMENTED** [review]
Looks good, the only other thing I was thinking is we can also replace the `CrossMarginUnsupported` with this so we have a single component to handle this across both futures types.

File: sections/futures/Trade/FuturesUnsupported.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			)}
+		</MessageContainer>
+	);
+};
+
+const UnsupportedMessage = styled.div`
+	margin-top: 12px;
+`;
+
+const IsolatedLink = styled.div`
```
Comment (avclarke) [inline sections/futures/Trade/FuturesUnsupported.tsx:42]: I think we can rename this to just Link or something more generic.

Comment (LeifuChen) [reply sections/futures/Trade/FuturesUnsupported.tsx:42]: We use `Link` from `next/link` in the code repo. So maybe name this as `LinkContainer`? 
image

Comment (avclarke) [reply sections/futures/Trade/FuturesUnsupported.tsx:42]: Yea sounds good

Comment (LeifuChen) [reply sections/futures/Trade/FuturesUnsupported.tsx]: Done.

**Comment by LeifuChen** [comment]
> Looks good, the only other thing I was thinking is we can also replace the `CrossMarginUnsupported` with this so we have a single component to handle this across both futures types.

Sounds like a good idea. @platschi do you want to have a more generic component in the new PR or in this PR?

**Review by avclarke — APPROVED** [review]

**Comment by avclarke** [comment]
> > Looks good, the only other thing I was thinking is we can also replace the `CrossMarginUnsupported` with this so we have a single component to handle this across both futures types.
> 
> Sounds like a good idea. @platschi do you want to have a more generic component in the new PR or in this PR?

I don't mind updating the cross margin part actually as I have a couple of ideas to improve the state there in general.

**Pull request merged**